Library/ViewModels/SettingsNewslettersCellViewModel.swift

```diff
@@ -0,0 +1,181 @@
+import KsApi
+import Prelude
+import ReactiveSwift
+import Result
+
+public protocol SettingsNewslettersCellViewModelInputs {
+
+  func allNewslettersSwitchTapped(on: Bool)
+  func awakeFromNib()
+  func configureWith(value: (newsletter: Newsletter, user: User))
+  func newslettersSwitchTapped(on: Bool)
+}
+
+public protocol SettingsNewslettersCellViewModelOutputs {
+
+  var showOptInPrompt: Signal<String, NoError> { get }
+  var subscribeToAllSwitchIsOn: Signal<Bool?, NoError> { get }
+  var switchIsOn: Signal<Bool?, NoError> { get }
+  var unableToSaveError: Signal<String, NoError> { get }
+  var updateCurrentUser: Signal<User, NoError> { get }
+}
+
+public protocol SettingsNewslettersCellViewModelType {
+
+  var inputs: SettingsNewslettersCellViewModelInputs { get }
+  var outputs: SettingsNewslettersCellViewModelOutputs { get }
+}
+
+public final class SettingsNewsletterCellViewModel: SettingsNewslettersCellViewModelType,
+SettingsNewslettersCellViewModelInputs, SettingsNewslettersCellViewModelOutputs {
+
+  public init() {
+
+    let newsletter = self.newsletterProperty.signal.skipNil()
+
+    let initialUser = self.awakeFromNibProperty.signal
+      .flatMap {
+        AppEnvironment.current.apiService.fetchUserSelf()
+          .wrapInOptional()
+          .prefix(value: AppEnvironment.current.currentUser)
+          .demoteErrors()
+      }
+      .skipNil()
+      .skipRepeats()
+
+    let newsletterOn: Signal<(Newsletter, Bool), NoError> = newsletter
+      .takePairWhen(self.newslettersSwitchTappedProperty.signal.skipNil())
+      .map { newsletter, isOn in (newsletter, isOn) }
+
+    self.showOptInPrompt = newsletterOn
+      .filter { _, on in AppEnvironment.current.config?.countryCode == "DE" && on }
+      .map { newsletter, _ in newsletter.displayableName }
+
+    let userAttributeChanged: Signal<(UserAttribute, Bool), NoError> = Signal.combineLatest(
+        newsletter,
+        self.newslettersSwitchTappedProperty.signal.skipNil()
+    ).map { newsletter, isOn in
+      (UserAttribute.newsletter(newsletter), isOn)
+    }
+
+    let updatedUser = initialUser
+      .switchMap { user in
+        userAttributeChanged.scan(user) { user, attributeAndOn in
+          let (attribute, on) = attributeAndOn
+          return user |> attribute.lens .~ on
+        }
+    }
+
+    let updateUserAllOn = initialUser
+      .takePairWhen(self.allNewslettersSwitchProperty.signal.skipNil())
+      .map { user, on in
+        return user
+          |> User.lens.newsletters .~ User.NewsletterSubscriptions.all(on: on)
+    }
+
+    let updateEvent = Signal.merge(updatedUser, updateUserAllOn)
+      .switchMap {
+        AppEnvironment.current.apiService.updateUserSelf($0)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .materialize()
+    }
+
+    self.unableToSaveError = updateEvent.errors()
```

Comment:
What happens when this signal emits? I don't see any observers on `SettingsNewslettersCell`

Reply:
Has missed that one. It sends a message to the VC so it can show an error prompt to the user.